src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,102 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private T[] elements;
+
+    public ArrayList() {
+        elements = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    public ArrayList(int capacity) {
+        if (capacity > 0) {
+            elements = (T[]) new Object[capacity];
+        }
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elements.length) {
+            growArray();
+        }
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index is invalid - " + index);
+        }
+        if (size == elements.length) {
+            growArray();
+        }
+        System.arraycopy(elements, index, elements, index + 1, size - index);
+        elements[index] = value;
+        size++;
```

Comment:
```suggestion
        System.arraycopy(elements, index, elements, index + 1, size - index);
        elements[index] = value;
        size++;
```
по-суті тут достатньо одного system.arraycopy